src/overlays/effects/ovl_Effect_Ss_Hitmark/z_eff_ss_hitmark.c

```diff
@@ -1,27 +1,128 @@
 
-#if 0
+static Color_RGB8 sColors[] = {
+    { 255, 255, 255 }, { 255, 255, 0 }, { 255, 255, 255 }, { 255, 0, 0 },   { 255, 200, 100 }, { 200, 150, 0 },
+    { 150, 100, 0 },   { 100, 50, 0 },  { 255, 255, 255 }, { 255, 0, 0 },   { 255, 255, 0 },   { 255, 0, 0 },
+    { 255, 255, 255 }, { 0, 255, 200 }, { 255, 255, 255 }, { 150, 0, 255 },
+};
+
+static TexturePtr sTextures[] = {
+    gEffHitmark1Tex,  gEffHitmark2Tex,  gEffHitmark3Tex,  gEffHitmark4Tex,  gEffHitmark5Tex,  gEffHitmark6Tex,
+    gEffHitmark7Tex,  gEffHitmark8Tex,  gEffHitmark9Tex,  gEffHitmark10Tex, gEffHitmark11Tex, gEffHitmark12Tex,
+    gEffHitmark13Tex, gEffHitmark14Tex, gEffHitmark15Tex, gEffHitmark16Tex, gEffHitmark17Tex, gEffHitmark18Tex,
+    gEffHitmark19Tex, gEffHitmark20Tex, gEffHitmark21Tex, gEffHitmark22Tex, gEffHitmark23Tex, gEffHitmark24Tex,
+    gEffHitmark1Tex,  gEffHitmark2Tex,  gEffHitmark3Tex,  gEffHitmark4Tex,  gEffHitmark5Tex,  gEffHitmark6Tex,
+    gEffHitmark7Tex,  gEffHitmark8Tex,
+};
+
 const EffectSsInit Effect_Ss_Hitmark_InitVars = {
     EFFECT_SS_HITMARK,
     EffectSsHitmark_Init,
 };
 
-#endif
+u32 EffectSsHitmark_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsHitmarkInitParams* initParams = PARAMS;
+    s32 colorIndex;
+
+    this->pos = initParams->pos;
+    this->gfx = gEffHitmarkDL;
+
+    if (initParams->type == EFFECT_HITMARK_DUST) {
+        this->life = 16;
+    } else {
+        this->life = 8;
+    }
+
+    this->draw = EffectSsHitmark_Draw;
+    this->update = EffectSsHitmark_Update;
+
+    colorIndex = initParams->type * 4;
+    this->rTexIndex = 0;
+    this->rType = initParams->type;
+    this->rPrimColorR = sColors[colorIndex].r;
+    this->rPrimColorG = sColors[colorIndex].g;
+    this->rPrimColorB = sColors[colorIndex].b;
+    this->rEnvColorR = sColors[colorIndex + 1].r;
+    this->rEnvColorG = sColors[colorIndex + 1].g;
+    this->rEnvColorB = sColors[colorIndex + 1].b;
+    this->rScale = initParams->scale;
+
+    return 1;
+}
+
+void EffectSsHitmark_Draw(PlayState* play, u32 index, EffectSs* this) {
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+    MtxF mfTrans;
+    MtxF mfScale;
+    MtxF mfResult;
+    MtxF mfTransBillboard;
+    Mtx* mtx;
+    f32 scale;
+    TexturePtr tex;
+    s32 pad;
+
+    OPEN_DISPS(gfxCtx);
+
+    SkinMatrix_SetTranslate(&mfTrans, this->pos.x, this->pos.y, this->pos.z);
+    scale = this->rScale / 100.0f;
+    SkinMatrix_SetScale(&mfScale, scale, scale, 1.0f);
+    SkinMatrix_MtxFMtxFMult(&mfTrans, &play->billboardMtxF, &mfTransBillboard);
+    SkinMatrix_MtxFMtxFMult(&mfTransBillboard, &mfScale, &mfResult);
+    gSPMatrix(POLY_XLU_DISP++, &gIdentityMtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    mtx = SkinMatrix_MtxFToNewMtx(gfxCtx, &mfResult);
+
+    if (mtx != NULL) {
+        gSPMatrix(POLY_XLU_DISP++, mtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        tex = sTextures[(this->rType * 8) + (this->rTexIndex)];
```

Comment:
```suggestion
        tex = sTextures[this->rType][this->rTexIndex];
```
Could this match as a 2D array?
`static TexturePtr sTextures[4][8] = {`

Reply:
Unfortunately, No. Same as in OoT.